CONTRIBUTING.md

```diff
@@ -0,0 +1,213 @@
+- **Other information**: For example relevant error messages or logs
+
+**Template:**
+```markdown
+## Problem Description
+Brief description of the issue
+
+## Steps to Reproduce
+1. Step one
+2. Step two
+3. Step three
+
+## Current Behavior
+What actually happens
+
+## Expected Behavior
+What should happen
+
+## Other information
+Logs, error messages, etc.
+```
+
+### Feature Requests
+
+For feature requests, please provide:
+
+**Required Information:**
+- **Use Case**: Describe the problem you're trying to solve
+- **Proposed Solution**: Your suggested approach (if any)
+- **Alternatives Considered**: Other solutions you've thought about
+- **Impact**: Who would benefit from this feature
+
+**Template:**
+```markdown
+## Use Case
+Describe the problem or need this feature would address
+
+## Proposed Solution
+Your suggested implementation approach
+
+## Alternatives Considered
+Other potential solutions
+
+## Additional Context
+Any other relevant information
+```
+
+## How to Submit Pull Requests
+
+### Development Workflow
+
+1. **Fork and Clone**: Fork this repository and clone your fork
+2. **Create Feature Branch**: Create a new branch from `main`
+3. **Make Changes**: Implement your changes following other guidelines
+    - Some tasks (like Buildah) may have several variants (buildah, buildah-oci-ta, buildah-remote...). Make changes only to the base version, e.g. `buildah.yaml`
+4. **Generate Content**: Run generation scripts if needed
+   ```bash
+   ./hack/generate-everything.sh
+   ```
+6. **Commit Changes**: See [commit guidelines](#pull-request-guidelines)
+
+### Pull Request Guidelines
+
+**Commit Requirements:**
+- Write clear, descriptive commit titles. Should fit under 50 characters
+- Write meaningful commit descriptions with each line having less than 72 characters
+- Split your contribution into several commits if applicable, each should represent a logical chunk
+- Add line `Assisted-by: <name-of-ai-tool> AI` if you used an AI tool for your contribution
+
+**Pull Request Content:**
+- **Title**: Clear, descriptive title. Should fit under 50 characters.
+- **Description**: Explain the changes and their purpose. You may refer to commit descriptions.
+- **Testing**: Describe how the changes were tested
+- **Links**: Reference related issues or upstream stories.
+
+**Remember:**
+- Konflux is a community project and proper descriptions may not be replaced by referencing a publicly inaccessible link to Jira or any other private resource.
+- Reviewers, other contributors and future generations might not have the same context as you have at the moment of PR submission.
+
+### Code Quality Standards
+
+**Shell Scripts:**
+- Set `set -euo pipefail` at the beginning of bash scripts
+- Pass shellcheck
+- Explain unintuitive code with comments
+
+### Security Best Practices
+
+**Critical Requirements:**
+- Never commit secrets or keys to the repository
+- Never expose or log sensitive information
+
+## Contribution Types
```

Comment:
General feedback for this section: it duplicates some of the sections from README.md, but with an insufficient level of detail. I don't think an external contributor would know what to do based only on the instructions here.

I think it would make sense to move the necessary information from README.md to CONTRIBUTING.md. But IMO it's unnecessary to include this info at all for build-definitions, since many things will change after decentralization